torch_xla/experimental/xla_sharding.py

```diff
@@ -148,6 +148,12 @@ class ShardingSpec:
   mesh: Mesh
   partition_spec: Tuple[Union[int, None]]
 
+  def can_apply(self, t: torch.Tensor) -> bool:
+    """
+    Test whether the ShardingSpec is compatible with the given torch.Tensor.
+    """
+    return len(self.mesh.mesh_shape) == len(t.shape)
```

Comment:
@jonb377 this check is not correct, mesh rank is not constraint by the input rank (e.g., if mesh rank is 1, and we have a 2D input tensor, it will be linearly sharded in one direction). Use partition_spec instead, 
```
len(t.shape) >= len(partition_spec)
```

We will have more checks inside `mark_sharding` but those should not be used for compatibility, and should crash if not met.

Reply:
Done, thanks Yeounoh!